src/dosbox.cpp

```diff
@@ -194,16 +195,16 @@ void increaseticks() { //Make it return ticksRemain and set it in the function a
 		ticksAdded = 0;
 
 		if (!CPU_CycleAutoAdjust || CPU_SkipCycleAutoAdjust || sleep1count < 3) {
-			Delay(1);
+			use_delay_precise ? DelayPrecise(1) : Delay(1);
 		} else {
 			/* Certain configurations always give an exact sleepingtime of 1, this causes problems due to the fact that
 			   dosbox keeps track of full blocks.
 			   This code introduces some randomness to the time slept, which improves stability on those configurations
 			 */
-			static const Bit32u sleeppattern[] = { 2, 2, 3, 2, 2, 4, 2};
+			static const Bit32u sleeppattern[] = { 2, 2, 3, 2, 2, 4, 2 };
 			static Bit32u sleepindex = 0;
 			if (ticksDone != lastsleepDone) sleepindex = 0;
-			Delay(sleeppattern[sleepindex++]);
+			use_delay_precise ? DelayPrecise(sleeppattern[sleepindex++]) : Delay(sleeppattern[sleepindex++]);
```

Comment:
This `use_delay_precise ? ` branch-point will be evaluated 1,000 times a second; would be nice to make this is a one-time call. 

(see comment on line 365)

Reply:
This will be a good excuse for me to look into std::function vs. old-school function pointers